lib/logitech_receiver/notifications.py

```diff
@@ -202,9 +202,10 @@ def _process_hidpp10_notification(device, status, n):
 def _process_feature_notification(device, status, n, feature):
 	if feature == _F.BATTERY_STATUS:
 		if n.address == 0x00:
-			discharge = ord(n.data[:1])
-			battery_status = ord(n.data[1:2])
-			status.set_battery_info(discharge, _hidpp20.BATTERY_STATUS[battery_status])
+			discharge_level = ord(n.data[:1])
+			discharge_next_level = ord(n.data[1:2])
+			battery_status = ord(n.data[2:3])
+			status.set_battery_info(discharge_level, _hidpp20.BATTERY_STATUS[battery_status])
```

Comment:
In `lib/logitech_receiver/hidpp10.py`, function `parse_battery_status` this register is also parsed. Does that have to be changed as well? The interpretation seems different.

Reply:
Hmm, this function is used for hid++ 1.0 protocol, which is, I expect different. Sorry, don't have hid++ 1.0 devices in hand. 
But if you look in `lib/logitech_receiver/hidpp20.py` function `get_battery()` interpret returned data in same way.
Also, look at provided logs: 
here 
`DEBUG [ReceiverListener:hidraw3] logitech_receiver.base: (13) => r[11 01 040C 5A050000000000000000000000000000]
17:36:50,184    DEBUG [ReceiverListener:hidraw3] logitech_receiver.hidpp20: device 1 battery 90% charged, next level 5% charge, status 0 = discharging`
battery status parsed by function get_battery() correctly
But in notification data parsed wrong
` 17:36:50,186    DEBUG [ReceiverListener:hidraw3] logitech_receiver.status: <PairedDevice(1,400A,M325)>: battery 90, invalid battery
17:36:50,186    DEBUG [MainThread] solaar.ui: status changed: <PairedDevice(1,400A,M325)> (1) None
17:36:50,186  WARNING [ReceiverListener:hidraw3] logitech_receiver.status: <PairedDevice(1,400A,M325)>: battery 90%, ALERT invalid battery
17:36:50,188    ERROR [ReceiverListener:hidraw3] logitech_receiver.listener: processing Notification(1,04,00,5A050000000000000000000000000000) `

Also see battery status in `docs/logitech/logitech_hidpp_2.0_specification_draft_2012-06-04.pdf`